tests/v2/validation/fleet/fleet.go

```diff
@@ -0,0 +1,155 @@
+package fleet
+
+import (
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	extensionsfleet "github.com/rancher/shepherd/extensions/fleet"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	DeploymentResourceSteveType = "apps.deployment"
+	FleetClusterResourceType    = "fleet.cattle.io.cluster"
+	fleetControllerName         = "cattle-fleet-system/fleet-controller"
+)
+
+func verifyGitRepo(t *testing.T, client *rancher.Client, gitRepoID, k8sClusterID, steveClusterID string) {
```

Comment:
Maybe, but I couldn't think of use cases outside of this integration package where fleet would be used in rancher repo. So it didn't make sense at this time for it to be an action, but maybe I'm wrong?

Reply:
I'm thinking more from backup restore cases. if we want to include fleet scenarios, and if this is maintained in actions, we could leverage this function instead of re writing it for BRO tests again.

Reply:
I can enhance this / move it to actions as we grow this folder of tests but with it being the first one, I think it makes the most sense to leave it here. I can't really tell what will change with the validation  /how reusable this function as is until we add more tests.